barretenberg/cpp/pil/avm/avm_alu.pil

```diff
@@ -622,3 +622,116 @@ namespace avm_alu(256);
+    // Since we need a range check we use the existing a_lo column that is range checked over 128-bits
+    op_div_a_lt_b * (a_lo - (ib - ia - 1)) = 0;
+    op_div_a_lt_b * ic  = 0; // ic = 0
+    op_div_a_lt_b * (ia - remainder) = 0; // remainder = a, might not be needed.
+
+
+    // ====== Handling ia > ib =====
+    pol commit op_div_std;
+    op_div_std * (1 - op_div_std) = 0;
+    pol commit divisor_lo; // b
+    pol commit divisor_hi;
+    op_div_std * (ib - divisor_lo - 2**64 * divisor_hi) = 0;
+    pol commit quotient_lo; // c
+    pol commit quotient_hi;
+    op_div_std * (ic - quotient_lo - 2**64 * quotient_hi) = 0;
+
+    // Multiplying the limbs gives us the following relations.
+    // (1) divisor_lo * quotient_lo --> Represents the bottom 128 bits of the result, i.e. values between [0, 2**128).
+    // (2) divisor_lo * quotient_hi + quotient_lo * divisor_hi --> Represents the middle 128 bits of the result, i.e. values between [2**64, 2**196)
+    // (3) divisor_hi * quotient_hi --> Represents the topmost 128 bits of the result, i.e. values between [2**128, 2**256).
+
+    // We simplify (2) by further decomposing it two limbs of 64 bits and adding the upper 64 bit to (3)
+    // divisor_lo * quotient_hi + quotient_lo * divisor_hi = partial_prod_lo + 2**64 * partial_prod_hi
+    // Need to range check that these are 64 bits
+    pol commit partial_prod_lo;
+    pol commit partial_prod_hi;
+    divisor_hi * quotient_lo + divisor_lo * quotient_hi = partial_prod_lo + 2**64 * partial_prod_hi;
+
+    pol PRODUCT = divisor_lo * quotient_lo + 2**64 * partial_prod_lo + 2**128 * (partial_prod_hi + divisor_hi * quotient_hi);
+
+    // a_lo and a_hi contains the hi and lo limbs of PRODUCT
+    // p_sub_a_lo and p_sub_a_hi contain the primality checks
+    #[ALU_PROD_DIV]
+    op_div_std * (PRODUCT - (a_lo + 2 ** 128 * a_hi)) = 0;
+    // Range checks already performed via a_lo and a_hi
+    // Primality checks already performed above via p_sub_a_lo and p_sub_a_hi
+
+    //Range check remainder < ib and put the value in b_hi, it has to fit into a 128 bit range check
+    #[REMAINDER_RANGE_CHK]
+     op_div_std * (b_hi - (ib - remainder - 1)) = 0;
+    
+    // We need to perform 3x 256-bit range checks: (a_lo, a_hi), (b_lo, b_hi), and (p_sub_a_lo, p_sub_a_hi) 
+    // One range check happens in-line with the division
+    #[CMP_CTR_REL_3]
+    (cmp_rng_ctr' - 2) * op_div_std = 0;
+    
+    // If we have more range checks left we cannot do more divisions operations that might truncate the steps
+    rng_chk_sel * op_div_std = 0;
+
+     // Check PRODUCT = ia - remainder
+    #[DIVISION_RELATION]
+    op_div_std * (PRODUCT - (ia - remainder)) = 0;
+
+    // === DIVISION 64 BIT RANGE CHECKS
+    // 64-bit decompositions and implicit 64 bit range checks for each limb,
+    // TODO: We need extra slice registers because we are performing an additional 64-bit bit range check in the same row, look into re-using old columns or refactoring
```

Comment:
```suggestion
    // TODO: We need extra slice registers because we are performing an additional 64-bit range check in the same row, look into re-using old columns or refactoring
```